lib/node/nodes/buffer.js

```diff
@@ -122,17 +138,11 @@ Buffer.prototype.isolinesDissolvedQuery = function() {
 };
 
 Buffer.prototype.checkLimits = function(context, callback) {
-    var defaultLimit;
-    var limitName;
-    var limit;
+    var limits = this.dissolved ? LIMITS_DISSOLVED : LIMITS;
 
-    if (this.dissolved) {
-        limitName = 'maximumNumberOfRowsDissolved';
-        defaultLimit = 100000;
-    } else {
-        limitName = 'maximumNumberOfRows';
-        defaultLimit = 1000000;
-    }
-    limit = context.getLimit(TYPE, limitName, defaultLimit, 'too many source rows');
-    limits.limitInputRows(this, 'source', context, limit, callback);
+    Checks.check(this, context, [{
+        checker: Checks.limitInputRows,
```

Comment:
The reason for the duality (`Checks.check` & `Checks.limitInputRows`) and the DSLish approach here is that `Checks.check` simplifies the processing if we want to combine multiple separate checks (e.g. check input rows and output rows), and handling for each separate check its limits values.

Then, for each check passed to `Checks.check`, the user can supply a `checker` function or use a predefined one such as `Checks.limitInputRows`here.

I see your point with the more object-oriented approach you propose, but I think we would miss the ability to easily have multiple checks. (which may not be important relly, I'm not sure... so far we're not needing it).